### PR TITLE
OCaml - update 1.7 snippets

### PR DESCRIPTION
**Pull request by ArulselvanMadhavan**
@pspencil pointed out that the snippets in chapter 7 were off by one. So, this PR fixes it.

@pspencil Let me know if this build works for you

### CODE REVIEW

File: src/content/1.7/code/ocaml/snippet03.ml [hunk 1]
```diff
@@ -1,6 +1 @@
-module type AtoB = sig
-  type a
-  type b
-
-  val f : a -> b
-end
+let f' f = function | None -> None | Some x -> Some (f x)
```
Comment (pspencil) [inline src/content/1.7/code/ocaml/snippet03.ml:1]: Shall we add linebreaks slimiar to previous version?

File: src/content/1.7/code/ocaml/snippet12.ml [hunk 2]
```diff
@@ -1 +1,4 @@
-type point = Pt of float * float
+module Point_Eq(E:Eq with type a = float) = struct
+  type a = point
+  let (==) (Pt (p1x, p1y)) (Pt (p2x, p2y)) = E.(p1x == p2x) && E.(p2x == p2y)
```
Comment (pspencil) [inline src/content/1.7/code/ocaml/snippet12.ml:3]: similarly, linebreaks?

File: src/content/1.7/code/ocaml/snippet15.ml [hunk 3]
```diff
@@ -1,8 +1 @@
-module Option_Functor : Functor with type 'a t = 'a option = struct
-  type 'a t = 'a option
-
-  let fmap f = function
-    | None -> None
-    | Some x -> Some (f x)
-  ;;
-end
+type 'a list = Nil | Cons of 'a * 'a list
```
Comment (pspencil) [inline src/content/1.7/code/ocaml/snippet15.ml:1]: Optional: linebreaks.

File: src/content/1.7/code/ocaml/snippet31.ml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 let fmapO = Option_Functor.fmap
 let fmapL = List_Functor.fmap
 let fmapC f l = (compose fmapO fmapL) f l
-let mis2 = fmapC square mis
+let mis2 = fmapC (square) mis
```
Comment (pspencil) [inline src/content/1.7/code/ocaml/snippet31.ml:4]: why did we add brackets here?

**Review by pspencil — COMMENTED** [review]
I can confirm that the matching of code is correct now. However, some line breaks and empty lines (e.g., the one between module type declaration and module elements) seem to be removed. 

I left a few comments on the ones that I think could be more readable with the line breaks.

File: src/content/1.7/code/ocaml/snippet15.ml (hunk 3 above)
Comment (ArulselvanMadhavan) [reply src/content/1.7/code/ocaml/snippet15.ml]: done

File: src/content/1.7/code/ocaml/snippet31.ml (hunk 4 above)
Comment (ArulselvanMadhavan) [reply src/content/1.7/code/ocaml/snippet31.ml]: removed

File: src/content/1.7/code/ocaml/snippet12.ml (hunk 2 above)
Comment (ArulselvanMadhavan) [reply src/content/1.7/code/ocaml/snippet12.ml]: Done

**Comment by ArulselvanMadhavan** [comment]
@pspencil Appreciate you catching that. It's been a while since I updated snippets that I forgot to run ocamlformat on them before sending the PR.

**Comment by pspencil** [comment]
No problem! I have been self studying this and your code really helped me a lot!

**Comment by ArulselvanMadhavan** [comment]
@hmemcpy Could you merge this when you get some time? Just has some formatting changes in Chapter 7 for the OCaml version

**Comment by hmemcpy** [comment]
Of course! Very sorry for neglecting this.

**Pull request merged**